src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -0,0 +1,158 @@
+import { useQuery } from '@apollo/client';
+import type { Profile } from '@generated/types';
+import { ProfilesDocument } from '@generated/types';
+import isFeatureEnabled from '@lib/isFeatureEnabled';
+import type { Conversation, Message, Stream } from '@xmtp/xmtp-js';
+import { SortDirection } from '@xmtp/xmtp-js';
+import { Client } from '@xmtp/xmtp-js';
+import { useEffect, useState } from 'react';
+import { useAppStore } from 'src/store/app';
+import { useMessageStore } from 'src/store/message';
+import { useSigner } from 'wagmi';
+
+const useMessagePreviews = () => {
+  const { data: signer } = useSigner();
+  const currentProfile = useAppStore((state) => state.currentProfile);
+  const client = useMessageStore((state) => state.client);
+  const setClient = useMessageStore((state) => state.setClient);
+  const isMessagesEnabled = isFeatureEnabled('messages', currentProfile?.id);
+
+  const [stream, setStream] = useState<Stream<Conversation>>();
+  const conversations = useMessageStore((state) => state.conversations);
+  const setConversations = useMessageStore((state) => state.setConversations);
+  const messageProfiles = useMessageStore((state) => state.messageProfiles);
+  const setMessageProfiles = useMessageStore((state) => state.setMessageProfiles);
+  const previewMessages = useMessageStore((state) => state.previewMessages);
+  const setPreviewMessages = useMessageStore((state) => state.setPreviewMessages);
+  const [messagesLoading, setMessagesLoading] = useState<boolean>();
+
+  const peerAddresses = Array.from(conversations.keys());
+  const request = { ownedBy: peerAddresses, limit: 50 };
+  const {
+    loading: profilesLoading,
+    error: profilesError,
+    fetchMore
+  } = useQuery(ProfilesDocument, {
+    variables: {
+      request: request
+    },
+    skip: !currentProfile?.id || peerAddresses.length === 0,
+    onCompleted: async (data) => {
+      if (!data?.profiles?.items?.length) {
+        return;
+      }
+      const profiles = data.profiles.items as Profile[];
+      const pageInfo = data.profiles.pageInfo;
+      const newMessageProfiles = new Map(messageProfiles);
+      for (const profile of profiles) {
+        const peerAddress = (profile.ownedBy as string).toLowerCase();
+        const previousProfile = newMessageProfiles.get(peerAddress);
+        if (!previousProfile?.isDefault) {
+          newMessageProfiles.set(peerAddress, profile);
+        }
+      }
+      setMessageProfiles(newMessageProfiles);
+
+      // Paginate through all profiles for the existing conversations.
+      if (pageInfo.next) {
+        await fetchMore({
+          variables: { request: { ...request, cursor: pageInfo?.next } }
+        });
+      }
+    }
+  });
+
+  useEffect(() => {
```

Comment:
I think this should be put into a new hook of its own. The initClient part

Reply:
I like that -- updated!